src/models/extra_inputs.cpp

```diff
@@ -5,6 +5,46 @@
 
 namespace Generators {
 
+PresetExtraInputs::PresetExtraInputs(State& state)
+    : state_(state),
+      registry_{
+          {"num_logits_to_keep", [&state = state_]() -> std::unique_ptr<OrtValue> {
+             std::vector<int64_t> shape{1};
+             auto num_logits_to_keep = OrtValue::CreateTensor(state.model_.allocator_cpu_, shape, ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64);
+             *num_logits_to_keep->GetTensorMutableData<int32_t>() = 0;
+             return num_logits_to_keep;
+           }}} {}
+
+void PresetExtraInputs::Add() {
+  const auto input_names_vector = state_.model_.session_info_->GetInputNames();
```

Comment:
Would it be simpler to just do a 'session_info_->HasInput("num_logits_to_keep")' and add it? There's already a special case for onnx::Neg_ prefixes.

Reply:
I think the logic currently added is more general purpose since it checks for all inputs that have not been populated already. For the next `preset extra input`, we would only need to add it to the registry.

Also, I don't want to add num_logits_to_keep if the user has supplied this input using params.set_input.